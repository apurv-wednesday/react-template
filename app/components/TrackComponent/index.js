/**
 *
 * TrackComponent
 *
 */

import React, { useState } from 'react';
import PropTypes from 'prop-types';
import styled from '@emotion/styled';
import Box from '@mui/material/Box';
import Card from '@mui/material/Card';
import CardContent from '@mui/material/CardContent';
import CardMedia from '@mui/material/CardMedia';
import IconButton from '@mui/material/IconButton';
import Typography from '@mui/material/Typography';
import PlayArrowIcon from '@mui/icons-material/PlayArrow';
import Pause from '@mui/icons-material/Pause';

const CustomCard = styled(Card)`
  && {
    margin: 1rem 0;
    padding: 1rem;
  }
`;
// eslint-disable-next-line require-jsdoc
export function TrackComponent({ trackName, artistName, previewUrl, artworkUrl100, collectionName }) {
  // console.log("component",data)
  const [isPlaying, setIsPlaying] = useState(false);
  const [audio] = useState(new Audio(previewUrl));
  const handlePlayClick = () => {
    if (isPlaying) {
      audio.pause();
    } else {
      audio.play();
    }
    setIsPlaying(!isPlaying);
  };
  return (
    <CustomCard sx={{ display: 'flex' }} data-testid="track-component">
      <Box sx={{ display: 'flex', flexDirection: 'column' }}>
        <CardContent sx={{ flex: '1 0 auto' }}>
          <Typography component="div" variant="h5">
            {trackName}
          </Typography>
          <Typography variant="subtitle1" color="text.secondary" component="div">
            {artistName}
          </Typography>
        </CardContent>
        <Box sx={{ display: 'flex', alignItems: 'center', pl: 1, pb: 1 }}>
          <IconButton aria-label="play/pause" onClick={handlePlayClick}>
            {isPlaying ? <Pause /> : <PlayArrowIcon />}
          </IconButton>
        </Box>
      </Box>
      <CardMedia component="img" sx={{ width: 151 }} image={artworkUrl100} alt={collectionName} />
      {isPlaying && <audio src={previewUrl} autoPlay />}
    </CustomCard>
  );
}

TrackComponent.propTypes = {
  trackName: PropTypes.string,
  previewUrl: PropTypes.string,
  artistName: PropTypes.string,
  artworkUrl100: PropTypes.string,
  collectionName: PropTypes.string
};

export default TrackComponent;
// functions : songName, songURL, artist(Can be a seperate component), play/pause
