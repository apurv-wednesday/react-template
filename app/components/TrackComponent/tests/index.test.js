/**
 *
 * Tests for TrackComponent
 *
 */

import React from 'react';
// import { fireEvent } from '@testing-library/dom'
import { renderWithIntl } from '@utils/testUtils';
import TrackComponent from '../index';

describe('<TrackComponent />', () => {
  it('should render and match the snapshot', () => {
    const { baseElement } = renderWithIntl(<TrackComponent />);
    expect(baseElement).toMatchSnapshot();
  });

  it('should contain 1 TrackComponent component', () => {
    const { getAllByTestId } = renderWithIntl(<TrackComponent />);
    expect(getAllByTestId('track-component').length).toBe(1);
  });
});
