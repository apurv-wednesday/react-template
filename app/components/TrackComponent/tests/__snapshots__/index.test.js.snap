// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TrackComponent /> should render and match the snapshot 1`] = `
<body>
  <div>
    <div
      class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root e1a3jrlg0 css-s4xvv6-MuiPaper-root-MuiCard-root"
      data-testid="track-component"
    >
      <div
        class="MuiBox-root css-j7qwjs"
      >
        <div
          class="MuiCardContent-root css-1qaznj3-MuiCardContent-root"
        >
          <div
            class="MuiTypography-root MuiTypography-h5 css-ag7rrr-MuiTypography-root"
          />
          <div
            class="MuiTypography-root MuiTypography-subtitle1 css-ecpxie-MuiTypography-root"
          />
        </div>
        <div
          class="MuiBox-root css-1ndgitk"
        >
          <button
            aria-label="play/pause"
            class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-78trlr-MuiButtonBase-root-MuiIconButton-root"
            tabindex="0"
            type="button"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
              data-testid="PlayArrowIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M8 5v14l11-7z"
              />
            </svg>
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
        </div>
      </div>
      <img
        class="MuiCardMedia-root MuiCardMedia-media MuiCardMedia-img css-1rorjgf-MuiCardMedia-root"
      />
    </div>
  </div>
</body>
`;
